set.mm

```diff
@@ -505661,6 +505729,150 @@ required as (atomic) formulas, can be introduced as a defined notion in
       $.
   $}
 
+  $( Lemma 1 for ~ satfv1lem .  (Contributed by AV, 17-Nov-2023.) $)
+  satfv1lemlem1 $p |- ( ( N e. V /\ F e. ( M ^m V ) /\ Z e. M )
+              -> ( ( { <. N , Z >. } u. ( F |` ( V \ { N } ) ) ) ` N ) = Z ) $=
+    ( wcel cmap co w3a cop csn cdif cres cun cfv wfn cin c0 wceq 3adant2 difssd
+    fnsng 3ad2ant2 fnssresd wn neldifsnd incom eqeq1i disjsn bitri sylibr snidg
+    elmapfn 3ad2ant1 fvun1 syl112anc fvsng eqtrd ) CDFZABDGHFZEBFZIZCCEJKZADCKZ
+    LZMZNOZCVCOZEVBVCVDPZVFVEPVDVEQZRSZCVDFZVGVHSUSVAVIUTCEDBUBTVBDVEAUTUSADPVA
+    ABDUMUCVBDVDUAUDVBCVEFUEZVKVBCDUFVKVEVDQZRSVMVJVNRVDVEUGUHVECUIUJUKUSUTVLVA
+    CDULUNVDVEVCVFCUOUPUSVAVHESUTCEDBUQTUR $.
+
+  $( Lemma 2 for ~ satfv1lem .  (Contributed by AV, 17-Nov-2023.) $)
+  satfv1lemlem2 $p |- ( ( ( N e. V /\ J e. V /\ -. J = N )
+                          /\ F e. ( M ^m V ) /\ Z e. M )
+      -> ( ( { <. N , Z >. } u. ( F |` ( V \ { N } ) ) ) ` J ) = ( F ` J ) ) $=
+    ( wcel wceq wn w3a cmap co cop csn cfv wfn cin c0 wa sylibr cdif cres simp1
+    cun anim1i 3adant2 fnsng syl elmapfn 3ad2ant2 difssd neldifsnd incom eqeq1i
+    fnssresd disjsn bitri neqne anim2i 3adant1 3ad2ant1 eldifsn fvun2 syl112anc
+    wne fvresd eqtrd ) DEGZBEGZBDHIZJZACEKLGZFCGZJZBDFMNZAEDNZUAZUBZUDOZBVROZBA
+    OVNVOVPPZVRVQPVPVQQZRHZBVQGZVSVTHVNVHVMSZWAVKVMWEVLVKVHVMVHVIVJUCUEUFDFECUG
+    UHVNEVQAVLVKAEPVMACEUIUJVNEVPUKUOVNDVQGIZWCVNDEULWCVQVPQZRHWFWBWGRVPVQUMUNV
+    QDUPUQTVNVIBDVEZSZWDVKVLWIVMVIVJWIVHVJWHVIBDURUSUTVABEDVBTZVPVQVOVRBVCVDVNB
+    VQAWJVFVG $.
```

Comment:
It looks like we already have `fvsnun` and `fvsnun2` which are very similar and use less hypotheses. I'm wondering if they could have been used here instead.

Also, as a note, elsewhere, modifying a function by changing its value at one point is formalized using `if`, like in `cantnfp1.f`, `symextf.e`, `df-limc`, `fourierdlem32.y`, `fdmdifeqresdif`. Should we recommend one vs. the other? (I believe the `if` works with proper classes, the function restriction not)

Reply:
> It looks like we already have fvsnun and fvsnun2 which are very similar and use less hypotheses. I'm wondering if they could have been used here instead.

You are right, fvsnun and fvsnun2 can be used directly in satfv1lem, so the lemmata satfv1lemlem1 and satfv1lemlem2 are not required/useful and can be removed. I will do this in my next commit.

Reply:
> Also, as a note, elsewhere, modifying a function by changing its value at one point is formalized using `if`, like in `cantnfp1.f`, `symextf.e`, `df-limc`, `fourierdlem32.y`, `fdmdifeqresdif`. Should we recommend one vs. the other? (I believe the `if` works with proper classes, the function restriction not)

This comes from the definition df-sat:
``Sat = ( m e. _V , e e. _V |->...  y = { a e. ( m ^m _om ) | A. z e. m ( { <. i , z >. } u.  ( a |` ( _om \ { i } ) ) ) e. ( 2nd ` u ) } ...``

Since we have no function values here, I think using an `if` expression would make it more complicated.